lib/SILOptimizer/Utils/Differentiation/ADContext.cpp

```diff
@@ -0,0 +1,112 @@
+//===--- ADContext.cpp - Context for Differentiation ----------*- C++ -*---===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// SWIFT_ENABLE_TENSORFLOW
+//
+// Per-module contextual information for the Differentiation pass.
+//===----------------------------------------------------------------------===//
+
+#define DEBUG_TYPE "differentiation"
+
+#include "swift/AST/DiagnosticsSIL.h"
+#include "swift/AST/Expr.h"
+#include "swift/SILOptimizer/PassManager/Transforms.h"
+#include "swift/SILOptimizer/Utils/Differentiation/ADContext.h"
+
+using llvm::DenseMap;
+using llvm::SmallPtrSet;
+using llvm::SmallVector;
+
+namespace swift {
+namespace autodiff {
+
+//===----------------------------------------------------------------------===//
+// Local Helpers
```

Comment:
(minor, feel free to address later)

Please capitalize only the first word in headings, for consistency with headings in other files.
```suggestion
// Local helpers
```